ledger-service/http-json/src/it/scala/http/HttpServiceIntegrationTest.scala

```diff
@@ -66,6 +78,149 @@ abstract class HttpServiceIntegrationTest
+        initialTplId = domain.TemplateId(None, "IIou", "TestIIou"),
+        // whether we can exercise by interface-ID
+        exerciseBy = iiouIfaceID,
+      )
+      // ideally we would upload IIou.daml only above, then upload ciou here;
+      // however tests currently don't play well with reload -SC
+      // next, use CIou
+      _ <- createIouAndExerciseTransfer(
+        initialTplId = domain.TemplateId(None, "CIou", "CIou"),
+        // whether we can exercise inherited by concrete template ID
+        exerciseBy = domain.TemplateId(None, "CIou", "CIou"),
+      )
+    } yield succeed
+  }
+
+  "fail to exercise by key with interface ID" in withHttpService { (uri, encoder, _, _) =>
+    import json.JsonProtocol._
+    for {
+      _ <- uploadPackage(uri)(ciouDar)
+      aliceH <- getUniquePartyAndAuthHeaders(uri)("Alice")
+      (alice, aliceHeaders) = aliceH
+      createTest <- postCreateCommand(
+        iouCommand(alice, domain.TemplateId(None, "CIou", "CIou")),
+        encoder,
+        uri,
+        aliceHeaders,
+      )
+      _ = createTest._1 should ===(StatusCodes.OK)
+      bobH <- getUniquePartyAndAuthHeaders(uri)("Bob")
+      (bob, _) = bobH
+      exerciseTest <- postJsonRequest(
+        uri withPath Uri.Path("/v1/exercise"),
+        encodeExercise(encoder)(
+          iouTransfer(
+            domain.EnrichedContractKey(
+              iiouIfaceID,
+              v.Value(v.Value.Sum.Party(domain.Party unwrap alice)),
+            ),
+            bob,
+          )
+        ),
+        aliceHeaders,
+      )
+    } yield {
+      val Status = StatusCodes.BadRequest
+      discard { exerciseTest._1 should ===(Status) }
+      inside(exerciseTest._2.convertTo[domain.ErrorResponse]) {
+        case domain.ErrorResponse(Seq(lookup), None, Status) =>
+          lookup should include regex raw"Cannot resolve Template Key type, given: TemplateId\([0-9a-f]{64},IIou,IIou\)"
+      }
+    }
+  }
+
+  // TODO(SC #13301) test against DB too
+  "fail to query by interface ID" in withHttpService { (uri, encoder, _, _) =>
+    import json.JsonProtocol._, spray.json.{enrichAny => `sj enrichAny`}
+    for {
+      _ <- uploadPackage(uri)(ciouDar)
+      aliceH <- getUniquePartyAndAuthHeaders(uri)("Alice")
+      (alice, aliceHeaders) = aliceH
+      searchResp <- search(
+        List.empty,
+        Map(
+          "templateIds" -> Seq(iiouIfaceID).toJson,
+          "query" -> spray.json.JsObject(),
+        ).toJson.asJsObject,
+        uri,
+        encoder,
+        aliceHeaders,
+      )
+    } yield inside(searchResp) {
+      case domain.ErrorResponse(Seq(_), None, StatusCodes.InternalServerError) => succeed
+    }
+  }
+
+  private[this] val (_, ciouVA) = {
+    import shapeless.syntax.singleton._
+    val iouT = Symbol("issuer") ->> VA.party ::
+      Symbol("owner") ->> VA.party ::
+      Symbol("amount") ->> VA.text ::
+      RNil
+    VA.record(Ref.Identifier assertFromString "none:Iou:Iou", iouT)
```

Comment:
Is this string `Iou:Iou` correct if we are interested in the `CIou`?

Reply:
It doesn't matter, this string could be `"foo:Bar:Baz"` for all the difference it makes to the test. So many conversions, one way or another.